components/formats-gpl/src/loci/formats/in/ZeissCZIReader.java

```diff
@@ -687,1706 +965,1576 @@ protected void initFile(String id) throws FormatException, IOException {
+
+    // Write all signatures
+    cziPartToSegments.forEach((part, cziSegments) -> { // For each part
+      Arrays.asList(cziSegments.subBlockDirectory.data.entries).forEach( // and each entry
+              entry -> {
+                int downscalingFactor = entry.getDimension("X").size/entry.getDimension("X").storedSize;
+                if ((downscalingFactor==1)||(allowAutostitching())) {
+                  // Split by resolution level if flattenedResolutions is true
+                  ModuloDimensionEntries moduloEntry = new ModuloDimensionEntries(entry,
+                          nRotations, nIlluminations, nPhases,
+                          nChannels, nSlices, nFrames, part);
+
+                  CoreSignature coreSignature = new CoreSignature(moduloEntry
+                          , RESOLUTION_LEVEL_DIMENSION,
+                          downscalingFactor,//getDownSampling(entry),
+                          maxDigitPerDimension::get,
+                          allowAutostitching(),
+                          FILE_PART_DIMENSION, part);
+                  if (!coreSignatureToBlocks.containsKey(coreSignature)) {
+                    coreSignatureToBlocks.put(coreSignature, new ArrayList<>());
+                  }
+                  coreSignatureToBlocks.get(coreSignature).add(moduloEntry);
+                }
+              });
+    });
+
+    // Sort them
+    List<CoreSignature> orderedCoreSignatureList = coreSignatureToBlocks.keySet().stream().sorted().collect(Collectors.toList());
+
+    // We now know how many core index are present in the image... except for extra images!
+
+    core = new ArrayList<>();
+    int idxCoreResolutionLevelStart = -1;
+    int idxSeries = -1;
+    // These variables (idxCoreResolutionLevelStart and idxSeries) roles are to keep track of the previous highest
+    // resolution level, and thus to abide by the rule nPixres0 = nPixres_i * downsampling_i
+    int previousMinX_maxRes = -1, previousMinY_maxRes = -1, previousPixX_maxRes = -1, previousPixY_maxRes = -1;
+    for (int iCore = 0; iCore<orderedCoreSignatureList.size(); iCore++) {
+      CoreMetadata core_i = new CoreMetadata();
+
+      //--------------- MODULO
+
+      // set modulo annotations
+      // rotations -> modulo Z
+      // illuminations -> modulo C
+      // phases -> modulo T
+
+      core_i.moduloZ.step = nSlices;
+      core_i.moduloZ.end = nSlices * (nRotations - 1);
+      core_i.moduloZ.type = FormatTools.ROTATION;
+
+      core_i.moduloC.step = nChannels;
+      core_i.moduloC.end = nChannels * (nIlluminations - 1);
+      core_i.moduloC.type = FormatTools.ILLUMINATION;
+      core_i.moduloC.parentType = FormatTools.CHANNEL;
+
+      core_i.moduloT.step = nFrames;
+      core_i.moduloT.end = nFrames * (nPhases - 1);
+      core_i.moduloT.type = FormatTools.PHASE;
+
+      //--------------- END OF MODULO
+
+      core.add(core_i);
+      core_i.orderCertain = true;
+      core_i.dimensionOrder = "XYCZT";
```

Comment:
this order seems to disagree with other areas including 'priority' which go XYZCT, is this on purpose?

Reply:
Sorry, I'm not sure I understand. Could you elaborate ? Which other areas ?